scripts/other/mteb_meta.py

```diff
@@ -0,0 +1,109 @@
+"""
+Usage: python mteb_meta.py path_to_results_folder
+
+Creates evaluation results metadata for the model card. 
+E.g.
+---
+tags:
+- mteb
+model-index:
+- name: SGPT-5.8B-weightedmean-msmarco-specb-bitfit
+  results:
+  - task:
+      type: classification
+    dataset:
+      type: mteb/banking77
+      name: MTEB Banking77
+      config: default
+      split: test
+    metrics:
+    - type: accuracy
+      value: 84.49350649350649
+---
+"""
+
+import io
+import json
+import os
+import sys
+
+from mteb import MTEB
+
+results_folder = sys.argv[1].strip("/")
+model_name = results_folder.split("/")[-1]
+
+all_results = {}
+
+for file_name in os.listdir(results_folder):
+    if not file_name.endswith(".json"):
+        print(f"Skipping non-json {file_name}")
+        continue
+    with io.open(os.path.join(results_folder, file_name), "r", encoding="utf-8") as f:
+        results = json.load(f)
+        all_results = {**all_results, **{file_name.replace(".json", ""): results}}
+
+MARKER = "---"
+TAGS = "tags:"
+MTEB_TAG = "- mteb"
+HEADER = "model-index:"
+MODEL = f"- name: {model_name}"
+RES = "  results:"
+
+META_STRING = "\n".join([MARKER, TAGS, MTEB_TAG, HEADER, MODEL, RES])
+
+
+ONE_TASK = "  - task:\n      type: {}\n    dataset:\n      type: {}\n      name: {}\n      config: {}\n      split: {}\n    metrics:"
+ONE_METRIC = "    - type: {}\n      value: {}"
+SKIP_KEYS = ["std", "evaluation_time", "main_score", "threshold"]
+
+for ds_name, res_dict in sorted(all_results.items()):
+    mteb_desc = (
+        MTEB(tasks=[ds_name.replace("CQADupstackRetrieval", "CQADupstackAndroidRetrieval")])
+        .tasks[0]
+        .description
+    )
+    hf_hub_name = mteb_desc.get("hf_hub_name", mteb_desc.get("beir_name"))
+    if "CQADupstack" in ds_name:
+        hf_hub_name = "BeIR/cqadupstack"
+    mteb_type = mteb_desc.get("type")
+    split = "test"
```

Comment:
That means we only report test scores? What if we want to report dev scores as well?

Reply:
Yeah we could also add all other split scores & filter for test scores in the space instead. 
The only downside is splits are not indicated in the sidebar, so it will be confusing if looking at the model card:
<img width="518" alt="Screenshot 2022-10-01 at 22 17 38" src="https://user-images.githubusercontent.com/62820084/193426873-41d1a77e-1dba-4b7d-9d0b-9a15f9befa4f.png">


Do you think we need other splits for any reason?

Reply:
Not really, I just thought it shouldnt be that difficult to add, since the library alrdy supports that. But yeah good point about the model card